### PR TITLE
kustomize-sops: init at 2.4.0

### PR DESCRIPTION
**Pull request by starcraft66**
###### Motivation for this change

There are not any [Kustomize](https://github.com/kubernetes-sigs/kustomize) plugins currently packaged in Nixpkgs yet.

Kustomize doesn't have a very robust plugin loading search path yet, it can only load plugins from a single "root" path which unfortunately means `buildEnv` must be used to merge multiple kustomize plugins into a single folder to make this package useful at the moment. Hopefully kustomize will eventually move to a plugin search path where multiple nix store paths can simply be appended to it in the future.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/kustomize/kustomize-sops.nix [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    sha256 = "0sr4d7amwn62xywwn83y58ynl8xv6l1q6zwbky5rmy0qxk909bqp";
+  };
+
+  vendorSha256 = "0vn6vrczbdln7ngz061xixjwn899jn7p2a46770xqx44bh3f2lgv";
+
+  postInstall = ''
+    mkdir -p $out/lib/viaduct.ai/v1/ksops-exec/
+    mv $out/bin/kustomize-sops $out/lib/viaduct.ai/v1/ksops-exec/ksops-exec
+    rm -r $out/bin
+  '';
```
Comment (SuperSandro2000) [inline pkgs/development/tools/kustomize/kustomize-sops.nix:20]:
```suggestion
  installPhase = ''
    mkdir -p $out/lib/viaduct.ai/v1/ksops-exec/
    mv $GOPATH/bin/kustomize-sops $out/lib/viaduct.ai/v1/ksops-exec/ksops-exec
  '';
```

File: pkgs/development/tools/kustomize/kustomize-sops.nix [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  meta = with lib; {
+    description = "A Flexible Kustomize Plugin for SOPS Encrypted Resource";
+    longDescription = ''
+      KSOPS can be used to decrypt any Kubernetes resource, but is most commonly
+      used to decrypt encrypted Kubernetes Secrets and ConfigMaps.
+    '';
+    homepage = "https://github.com/viaduct-ai/kustomize-sops";
+    license = licenses.asl20;
+    maintainers = with maintainers; [ starcraft66 ];
+    platforms = platforms.linux ++ platforms.darwin;
```
Comment (SuperSandro2000) [inline pkgs/development/tools/kustomize/kustomize-sops.nix:34]:
```suggestion
```

set by buildGoModule

File: pkgs/development/tools/kustomize/kustomize-sops.nix (hunk 1 above)
Comment (starcraft66) [reply pkgs/development/tools/kustomize/kustomize-sops.nix]: Thanks, done.

File: pkgs/development/tools/kustomize/kustomize-sops.nix (hunk 2 above)
Comment (starcraft66) [reply pkgs/development/tools/kustomize/kustomize-sops.nix]: Thanks, done.

**Comment by jlesquembre** [comment]
@starcraft66 maybe we can have a similar solution to the terraform derivation, something like `kustomize.withPlugins`

**Pull request merged**